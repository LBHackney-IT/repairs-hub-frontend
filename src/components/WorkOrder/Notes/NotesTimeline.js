import PropTypes from 'prop-types'
import NoteEntry from './NoteEntry'

const NotesTimeline = ({ notes }) => {
  return (
    <>
      {notes.length ? (
        <ul className="govuk-list note-timeline">
          {notes.map((note, index) => (
            <li
              key={index}
              className="note-entry govuk-body-s"
              data-note-id={index}
            >
              <NoteEntry key={index} {...note} />
            </li>
          ))}
        </ul>
      ) : (
        <p className="govuk-body-s">There are no notes for this work order.</p>
      )}
    </>
  )
}

NotesTimeline.propTypes = {
  notes: PropTypes.arrayOf(
    PropTypes.shape({
      note: PropTypes.string,
      user: PropTypes.string,
      time: PropTypes.instanceOf(Date),
    })
  ).isRequired,
}

export default NotesTimeline
