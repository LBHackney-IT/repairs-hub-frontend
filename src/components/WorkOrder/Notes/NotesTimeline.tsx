import { WorkOrder } from '@/root/src/models/workOrder'
import NoteEntry from './NoteEntry'
import { Note } from './types'
import { TabName } from '../../Tabs/tabNames'
import { WorkOrderAppointmentDetails } from '@/root/src/models/workOrderAppointmentDetails'

interface Props {
  notes: Note[]
  workOrder: WorkOrder
  appointmentDetails: WorkOrderAppointmentDetails
  setActiveTab: (tabName: TabName) => void
}

const NotesTimeline = (props: Props) => {
  const { notes, workOrder, setActiveTab, appointmentDetails } = props

  if (notes.length === 0) {
    return <p className="lbh-body-s">There are no notes for this work order.</p>
  }

  return (
    <ul className="lbh-list note-timeline">
      {notes.map((note, index) => {
        return (
          <li
            key={index}
            className="note-entry lbh-body-s"
            data-note-id={index}
          >
            <NoteEntry
              note={note}
              workOrder={workOrder}
              appointmentDetails={appointmentDetails}
              setActiveTab={setActiveTab}
            />
          </li>
        )
      })}
    </ul>
  )
}

export default NotesTimeline
