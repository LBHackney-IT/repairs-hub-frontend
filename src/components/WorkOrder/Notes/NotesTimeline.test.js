import { render } from '@testing-library/react'
import NotesTimeline from './NotesTimeline'

describe('NotesTimeline component', () => {
  const props = {
    notes: [
      {
        note: 'A more recent note about this repair',
        time: new Date('2021-02-08T15:06:40.057277'),
        user: 'Random Name',
        userEmail: 'random.name@hackney.gov.uk',
      },
      {
        note: 'A note about the repair',
        time: new Date('2021-02-08T15:05:12.743258'),
        user: 'Random Name',
        userEmail: 'random.name@hackney.gov.uk',
      },
    ],
  }

  it('should render properly when there are notes', () => {
    const { asFragment } = render(<NotesTimeline notes={props.notes} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render properly when there are no notes', () => {
    const { asFragment } = render(<NotesTimeline notes={[]} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
