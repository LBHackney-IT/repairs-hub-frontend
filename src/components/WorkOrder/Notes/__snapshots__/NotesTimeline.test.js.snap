// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesTimeline component should render properly when there are no notes 1`] = `
<DocumentFragment>
  <p
    class="govuk-body-s"
  >
    There are no notes for this work order.
  </p>
</DocumentFragment>
`;

exports[`NotesTimeline component should render properly when there are notes 1`] = `
<DocumentFragment>
  <ul
    class="govuk-list note-timeline"
  >
    <li
      class="note-entry govuk-body-s"
      data-note-id="0"
    >
      <div
        class="note-info govuk-body-s"
      >
        8 Feb 2021, 3:06 pm by Random Name (random.name@hackney.gov.uk)
      </div>
      A more recent note about this repair
    </li>
    <li
      class="note-entry govuk-body-s"
      data-note-id="1"
    >
      <div
        class="note-info govuk-body-s"
      >
        8 Feb 2021, 3:05 pm by Random Name (random.name@hackney.gov.uk)
      </div>
      A note about the repair
    </li>
  </ul>
</DocumentFragment>
`;
